tests/kubernetes/application/test_controller_wrapper.py

```diff
@@ -1,79 +1,290 @@
         app.kube_deployment.service.return_value = "fake_service"
         app.kube_deployment.instance.return_value = "fake_instance"
         Application.ensure_pod_disruption_budget(self=app, kube_client=mock_client)
-        mock_create_pdr.assert_called_once_with(
-            kube_client=mock_client, pod_disruption_budget=mock_req_pdr
+        mock_client.policy.create_namespaced_pod_disruption_budget.assert_called_once_with(
+            namespace="paasta", body=mock_req_pdr
         )
 
 
 def test_ensure_pod_disruption_budget_replaces_outdated():
     with mock.patch(
         "paasta_tools.kubernetes.application.controller_wrappers.pod_disruption_budget_for_service_instance",
         autospec=True,
-    ) as mock_pdr_for_service_instance, mock.patch(
-        "paasta_tools.kubernetes.application.controller_wrappers.create_pod_disruption_budget",
-        autospec=True,
-    ) as mock_create_pdr, mock.patch(
-        "paasta_tools.kubernetes.application.controller_wrappers.max_unavailable",
-        autospec=True,
-        return_value=0,
-    ):
+    ) as mock_pdr_for_service_instance:
         mock_req_pdr = mock.Mock()
-        mock_req_pdr.spec.min_available = 10
+        mock_req_pdr.spec.max_unavailable = 10
         mock_pdr_for_service_instance.return_value = mock_req_pdr
 
-        mock_client = mock.Mock()
+        mock_client = mock.MagicMock()
 
         mock_pdr = mock.Mock()
-        mock_pdr.spec.min_available = 10
+        mock_pdr.spec.max_unavailable = 5
 
-        mock_client.read_namespaced_pod_disruption_budget.return_value = mock_pdr
+        mock_client.policy.read_namespaced_pod_disruption_budget.return_value = mock_pdr
 
         app = mock.MagicMock()
-        app.soa_config.get_desired_instances.return_value = 10
+        app.soa_config.get_bounce_margin_factor.return_value = 10
         app.kube_deployment.service.return_value = "fake_service"
         app.kube_deployment.instance.return_value = "fake_instance"
         Application.ensure_pod_disruption_budget(self=app, kube_client=mock_client)
 
-        mock_client.policy.delete_namespaced_pod_disruption_budget.assert_called_once_with(
+        mock_client.policy.patch_namespaced_pod_disruption_budget.assert_called_once_with(
             name=mock_req_pdr.metadata.name,
             namespace=mock_req_pdr.metadata.namespace,
-            body=V1DeleteOptions(),
+            body=mock_req_pdr,
         )
-        mock_create_pdr.assert_called_once_with(
-            kube_client=mock_client, pod_disruption_budget=mock_req_pdr
+
+
+def test_sync_horizontal_pod_autoscaler():
+    mock_client = mock.MagicMock(autospec=True)
+    app = mock.MagicMock(autospec=True)
+    app.item.metadata.name = "fake_name"
+    app.item.metadata.namespace = "faasta"
+    mock_config = mock.MagicMock(autospec=True)
+
+    # helper Functions for mocking
+    def setup_app(app, config_dict, exists_hpa):
+        app.reset_mock()
```

Comment:
My preference is to split up tests into different test cases instead of resetting mocks, but I see this resetting pattern already exists in paasta_tools, so *shrug*

Reply:
Same as ^^